codecparsers/mpeg2_parser.cpp

```diff
@@ -356,12 +355,96 @@ namespace MPEG2 {
         DEBUG("time_code_minutes  : %x", m_GOPHeader.time_code_minutes);
         DEBUG("time_code_seconds  : %x", m_GOPHeader.time_code_seconds);
         DEBUG("time_code_pictures : %x", m_GOPHeader.time_code_pictures);
+        DEBUG("closed_gop 	  : %x", m_GOPHeader.closed_gop);
 
         bitReaderDeInit();
 
         return true;
     }
 
+    bool Parser::parseQuantMatrixExtension(const StreamHeader* shdr)
+    {
+        ExtensionIdentifierType extID;
+        const uint8_t* nalData = shdr->nalData;
+        int32_t nalSize = shdr->nalSize;
+        QuantMatrices *quantMatrices = &m_quantMatrixExtension.quantizationMatrices;
+        BitReader bitReader(nalData, nalSize);
+        bitReaderInit(&bitReader);
+        skipByte(); // skip start_sequence_code
+
+        memset(&m_quantMatrixExtension, 0, sizeof(QuantMatrixExtension));
+
+        // extension_start_code_identifier
+        bitReaderReadBits(
+            4, &(m_quantMatrixExtension.extension_start_code_identifier));
+
+        extID = static_cast<ExtensionIdentifierType>(
+            m_quantMatrixExtension.extension_start_code_identifier);
+
+        if (extID != kQuantizationMatrix) {
+            ERROR("Wrong extension id type");
+            bitReaderDeInit();
+            return false;
+        }
+
+        bitReaderReadFlag(&(quantMatrices->load_intra_quantiser_matrix));
```

Comment:
Could we use one function like
loadMatrix(bool& loaded, uint8_t matrix[64])
to remove 4 duplicate code below?